src/textual/renderables/bar.py

```diff
@@ -96,45 +106,25 @@ def __rich_console__(
         yield output_bar
 
 
-if __name__ == "__main__":
-    import random
-    from time import sleep
-
-    from rich.color import ANSI_COLOR_NAMES
-
-    console = Console()
-
-    def frange(start, end, step):
-        current = start
-        while current < end:
-            yield current
-            current += step
-
-        while current >= 0:
-            yield current
-            current -= step
-
-    step = 0.1
-    start_range = frange(0.5, 10.5, step)
-    end_range = frange(10, 20, step)
-    ranges = zip(start_range, end_range)
+def _apply_gradient(text: Text, gradient: Gradient, width: int) -> None:
+    """Apply a gradient to a Rich Text instance.
 
-    console.print(Bar(width=20), f"   (.0, .0)")
-
-    for range in ranges:
-        color = random.choice(list(ANSI_COLOR_NAMES.keys()))
-        console.print(
-            Bar(range, highlight_style=color, width=20),
-            f"   {range}",
+    Args:
+        text: A Text object.
+        gradient: A Textual gradient.
+        width: Width to apply gradient to, or None for full width.
+    """
+    from_color = Style.from_color
+    get_rich_color = gradient.get_rich_color
+
+    max_width = width - 1
+    if not max_width:
+        return
```

Comment:
If a renderable is resized down to width 1, should we maybe sample the "first" colour from the gradient instead of doing nothing? Feels like this would maybe flicker out.

Reply:
Good catch. Somebody will do that.